src/main/java/com/pingcap/tikv/policy/RetryPolicy.java

```diff
@@ -36,47 +39,53 @@
           Status.Code.UNIMPLEMENTED, Status.Code.OUT_OF_RANGE,
           Status.Code.UNAUTHENTICATED, Status.Code.CANCELLED);
 
-  public RetryPolicy(ErrorHandler handler) {
+  RetryPolicy(ErrorHandler<RespT> handler) {
     this.handler = handler;
   }
 
-  protected abstract boolean shouldRetry(Exception e);
+  private boolean checkNotRecoverableException(Status status) {
+    return unrecoverableStatus.contains(status.getCode());
+  }
 
-  protected boolean checkNotLeaderException(Status status) {
-    // TODO: need a way to check this, for now all unknown exception
-    return true;
+  private void handleFailure(Exception e, String methodName, long millis) {
+    Status status = Status.fromThrowable(e);
+    if (checkNotRecoverableException(status)) {
+      logger.error("Failed to recover from last grpc error calling %s.", methodName);
+      throw new GrpcException(e);
+    }
+    doWait(millis);
   }
 
-  protected boolean checkNotRecoverableException(Status status) {
-    return unrecoverableStatus.contains(status.getCode());
+  private void doWait(long millis) {
+    try {
+      Thread.sleep( backOff.nextBackOffMillis() );
+    } catch (InterruptedException e) {
+      throw new RuntimeException( e );
```

Comment:
Use a specific runtime error. All exceptions we defined are runtime exception error.

Reply:
DONE